worlds/sc2/client.py

```diff
@@ -718,15 +727,66 @@ def parse_mission_info(mission_info: dict[str, typing.Any]) -> MissionInfo:
         return MissionInfo(
             **{field: value for field, value in mission_info.items() if field in MissionInfo._fields}
         )
-
-    def find_campaign(self, mission_name: str) -> SC2Campaign:
-        data = self.mission_req_table
-        for campaign in data.keys():
-            if mission_name in data[campaign].keys():
-                return campaign
-        sc2_logger.info(f"Attempted to find campaign of unknown mission '{mission_name}'; defaulting to GLOBAL")
-        return SC2Campaign.GLOBAL
-
+    
+    @staticmethod
+    def parse_mission_req_table(mission_req_table: typing.Dict[SC2Campaign, typing.Dict[typing.Any, MissionInfo]]) -> typing.List[CampaignSlotData]:
+        campaigns: typing.List[typing.Tuple[int, CampaignSlotData]] = []
+        rolling_rule_id = 0
+        for (campaign, campaign_data) in mission_req_table.items():
+            if campaign.campaign_name == "Global":
+                campaign_name = ""
+            else:
+                campaign_name = campaign.campaign_name
+            
+            categories: typing.Dict[str, typing.List[MissionSlotData]] = {}
+            for mission in campaign_data.values():
+                if mission.category not in categories:
+                    categories[mission.category] = []
+                mission_id = mission.mission.id
+                sub_rules: typing.List[CountMissionsRuleData] = []
+                if mission.number:
+                    amount = mission.number
+                    missions = {
+                        mission.mission.id
+                        for mission in mission_req_table[campaign].values()
+                    }
+                    sub_rules.append(CountMissionsRuleData(missions, amount, [campaign_name]))
+                prev_missions: typing.Set[int] = set()
+                if len(mission.required_world) > 0:
+                    missions: typing.List[int] = []
+                    for connection in mission.required_world:
+                        if type(connection) == dict:
+                            con_campaign = {}
```

Comment:
Recommend using a full word instead of `con`, it's very hard to read even knowing it's supposed to be "connection". I can't tell at a glance whether it's the source or destination, so I recommend one of `from_`, `source_`, `to_`, `target_`, or `dest_` if you want destination but short (that one is standard in many Python builtins and libraries so more acceptable imo).

Two lines above, I also recommend `isinstance(connection, dict)` instead of `type(connection) == dict`; the former is the more standard way of doing it, so it's what readers expect. It's more expected because it also captures subclasses, which is unlikely to matter here but possible.

Reply:
Also recommend a type hint on this.